tensorflow/python/ops/check_ops.py

```diff
@@ -362,11 +385,22 @@ def _binary_assert(sym, opname, op_func, static_func, x, y, data, summarize,
         ]
       if message is not None:
         data = [message] + list(data)
-      condition = math_ops.reduce_all(op_func(x, y))
+      if allow_empty:
+        condition = math_ops.reduce_all(op_func(x, y))
+      else:
+        empty_check = _binary_all_empty_or_all_non_empty(x, y)
+        condition = math_ops.logical_and(
+            empty_check, math_ops.reduce_all(op_func(x, y)))
       x_static = tensor_util.constant_value(x)
       y_static = tensor_util.constant_value(y)
       if x_static is not None and y_static is not None:
-        condition_static = np.all(static_func(x_static, y_static))
+        if allow_empty:
+          condition_static = np.all(static_func(x_static, y_static))
+        else:
+          empty_check_static = ((x_static.size == 0 and y_static.size == 0) or
+                                (x_static.size != 0 and y_static.size != 0))
+          condition_static = empty_check_static and np.all(
+              static_func(x_static, y_static))
```

Comment:
`InvalidArgumentError: assertion failed: [`Tensor(\"Placeholder:0\", dtype=int32)` elements must be either positive integers or `-1`.] [Condition x >= y did not hold element-wise:] [x (Placeholder:0) = ] [] [y (reshape/assert_greater_equal/y:0) = ] [-1]
	 [[node reshape/assert_greater_equal/Assert/Assert (defined at /tensorflow/contrib/distributions/python/kernel_tests/bijectors/reshape_test.py:111) ]]

Errors may have originated from an input operation.
Input Source operations connected to node reshape/assert_greater_equal/Assert/Assert:
```suggestion
              static_func(x_static, y_static))` 
We can see the above internal error, can please fix it.